app/scenes/Document/components/PublishPopover.tsx

```diff
@@ -0,0 +1,227 @@
+import FuzzySearch from "fuzzy-search";
+import { uniq } from "lodash";
+import { observer } from "mobx-react";
+import * as React from "react";
+import { useTranslation, Trans } from "react-i18next";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { FixedSizeList as List } from "react-window";
+import styled from "styled-components";
+import Document from "~/models/Document";
+import Button from "~/components/Button";
+import Flex from "~/components/Flex";
+import { Outline } from "~/components/Input";
+import InputSearch from "~/components/InputSearch";
+import PublishLocation from "~/components/PublishLocation";
+import Text from "~/components/Text";
+import useStores from "~/hooks/useStores";
+import useToasts from "~/hooks/useToasts";
+import { flattenTree, ancestors } from "~/utils/tree";
+
+type Props = {
+  document: Document;
+  visible: boolean;
+};
+
+function PublishPopover({ document, visible }: Props) {
+  const [searchTerm, setSearchTerm] = React.useState<string>();
+  const [selectedLocation, setLocation] = React.useState<any>();
+  const { collections } = useStores();
+  const { showToast } = useToasts();
+  const { t } = useTranslation();
+  const listRef = React.useRef<List>(null);
+
+  React.useEffect(() => {
+    if (visible && selectedLocation) {
+      listRef.current?.scrollToItem(selectedLocation.index);
+    }
+  });
+
+  const searchIndex = React.useMemo(() => {
+    const data = flattenTree(collections.tree.root).slice(1);
```

Comment:
I assume we don't need to filter out templates here? From what I understood _Templatize_ option creates a copy and doesn't alter the existing `documentStructure`.